src/vsr/grid_scrubber.zig

```diff
@@ -156,6 +159,71 @@ pub fn GridScrubberType(comptime Forest: type) type {
             scrubber.* = undefined;
         }
 
+        pub fn open(scrubber: *GridScrubber, random: std.rand.Random) void {
+            // Compute the tour origin exactly once.
+            if (scrubber.tour_tables_origin) |_| {
+                return;
+            }
+
+            // Each replica's scrub origin is chosen independently.
+            // This reduces the chance that the same block across multiple replicas can bitrot
+            // without being discovered and repaired by a scrubber.
+            //
+            // To accomplish this, try to select an origin uniformly across all blocks:
+            // - Bias towards levels with more tables.
+            // - Bias towards trees with larger Values (since their tables have more blocks).
+            // - (Though, for ease of implementation, the origin is always at the beginning of a
+            //   tree's level, never in the middle.)
+            assert(scrubber.tour == .init);
+
+            const tree_count = Forest.tree_infos.len;
+
+            var weights: [tree_count][constants.lsm_levels]u64 = undefined;
+            var weights_sum: u64 = 0;
+            for (0..constants.lsm_levels) |level| {
+                inline for (Forest.tree_infos, 0..) |tree_info, tree_index| {
+                    const tree_id = comptime Forest.tree_id_cast(tree_info.tree_id);
+                    const tree = scrubber.forest.tree_for_id_const(tree_id);
+                    const levels = &tree.manifest.levels;
+                    weights[tree_index][level] =
+                        levels[level].tables.len() * @sizeOf(tree_info.Tree.Table.Value);
+                    weights_sum += levels[level].tables.len();
```

Comment:
Hm, this is not the sum of weights, no? I'd expect to see

```
weights_sum += weights[tree_index][level]
```

Reply:
Good catch!